### DISCUSSION TITLE
pushimage for 4bbp

### DISCUSSION THREAD

**Discussion by ESPboy-edu** [Bodmer/TFT_eSPI#1093] [Q&A - General] [answered]
using  
void pushImage(int32_t x, int32_t y, int32_t w, int32_t h, uint8_t  *data, bool bpp8 = true, uint16_t *cmap = nullptr);
for 4bbp as
tft.pushImage(16, 16, 16, 16, (uint8_t *)tile, false, bit4palette);
causes reset

but with sprite work ok
spr.createSprite(16, 16);
spr.createPalette(bit4palette); 
spr.pushImage(0, 0, 16, 16, (uint16_t *)tile);
spr.pushSprite(16, 16);

what's the problem could be?
i suppose that direct tft.pushImage could be faster than spr.pushImage + spr.pushSprite

**Comment by ESPboy-edu** [comment]
it was due to image in PROGMEM
if i putting image to RAM tft.pushImage(16, 16, 16, 16, (uint8_t *)tile, false, bit4palette); works fine!
how could I say to  tft.pushImage to work with PROGMEM )

**Comment by Bodmer** [comment, marked as answer]
See #1094
